c10/core/SingletonSymNodeImpl.h

```diff
@@ -0,0 +1,121 @@
+#include <c10/core/ConstantSymNodeImpl.h>
+#include <c10/core/SymBool.h>
+#include <c10/core/SymNodeImpl.h>
+#include <iostream>
+
+namespace c10 {
+
+// An object that only defines the equality operator.
```

Comment:
With itself!

A little more motivating explanation here is always helpful.

Reply:
Hmm but we do allow `singleton(1) == singleton(1)` and that seems more like normal equality to me?

We only get the "with itself" idea once we have something like `get_next_unique_singleton(offset) != get_next_unique_singleton(offset.clone())`.